Services.Test/InternalDevicePropertiesTest.cs

```diff
@@ -0,0 +1,222 @@
+            var key = "testSetKey";
+            var value = "testSetValue";
+
+            // Act
+            this.target.Set(key, value);
+            var result = this.target.Get(key);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(result, value);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void SetStateValueTest()
+        {
+            // Arrange
+            this.target = this.GetEmptyProperties();
+            var key = "testSetKey";
+            var value = "testSetValue";
+            this.target.Set(key, value);
+
+            // Act
+            var result = this.target.Get(key);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(result, value);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void HasPropertyReturnsTrueForProperty()
+        {
+            // Arrange
+            this.target = this.GetEmptyProperties();
+            var key = "testHasKey";
+            var value = "testHasValue";
+            this.target.Set(key, value);
+
+            // Act
+            var result = this.target.Has(key);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void HasPropertyReturnsFalseIfNotThere()
+        {
+            // Arrange
+            this.target = this.GetEmptyProperties();
+            var key = "KeyThatDoesNotExist";
+
+            // Act
+            var result = this.target.Has(key);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ChangedIsSetToTrueForNewProperty()
+        {
+            // Arrange
+            this.target = this.GetEmptyProperties();
+            Assert.False(this.target.Changed);
+
+            var key = "testKey";
+            var value = "testValue";
+
+            // Act
+            this.target.Set(key, value);
+
+            // Assert
+            Assert.True(this.target.Changed);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ResetChangedIsFalse()
+        {
+            // Arrange
+            this.target = this.GetTestProperties();
+
+            var key = "testKey";
+            var value = "testValue";
+
+            // Act
+            this.target.ResetChanged();
+
+            // Assert
+            Assert.False(this.target.Changed);
+        }
+
+        private InternalDeviceProperties GetEmptyProperties()
+        {
+            return new InternalDeviceProperties();
+        }
+
+        private InternalDeviceProperties GetTestProperties()
+        {
+            return new InternalDeviceProperties(GetTestChillerModel());
+        }
+
+        /// <summary>
+        /// Returns the a test chiller model
+        /// </summary>
+        private DeviceModel GetTestChillerModel()
```

Comment:
I understand it might be a matter of taste, but I'd suggest to avoid coupling data providers across tests. In my experience they increase the chance of failures when adding tests